data-raw/vs_peds.R

```diff
@@ -2,160 +2,218 @@
+      VISIT == "BASELINE" ~ 14.95,
+      VISIT == "WEEK 2" ~ 15.17,
+      VISIT == "WEEK 4" ~ 15.43,
+      VISIT == "WEEK 6" ~ 15.66,
+      VISIT == "WEEK 8" ~ 15.84,
+      VISIT == "WEEK 12" ~ 16.34,
+      VISIT == "WEEK 16" ~ 16.73,
+      VISIT == "WEEK 20" ~ 17.11,
+      VISIT == "WEEK 24" ~ 17.56,
+      VISIT == "WEEK 26" ~ 17.85
+    ),
+    USUBJID == "01-701-1028" & VSTESTCD == "HDCIRC" ~ case_when(
+      VISIT == "SCREENING 1" ~ 51.34,
+      VISIT == "BASELINE" ~ 51.71,
+      VISIT == "WEEK 2" ~ 52.12,
+      VISIT == "WEEK 4" ~ 52.56,
+      VISIT == "WEEK 6" ~ 52.93,
+      VISIT == "WEEK 8" ~ 53.45,
+      VISIT == "WEEK 12" ~ 54.02,
+      VISIT == "WEEK 16" ~ 54.48,
+      VISIT == "WEEK 20" ~ 54.97,
+      VISIT == "WEEK 24" ~ 55.44,
+      VISIT == "WEEK 26" ~ 55.82
+    ),
+    USUBJID == "01-701-1033" & VSTESTCD == "HEIGHT" ~ case_when(
+      VISIT == "SCREENING 1" ~ 59.45,
+      VISIT == "BASELINE" ~ 60.53,
+      VISIT == "WEEK 2" ~ 61.72,
+      VISIT == "WEEK 4" ~ 62.91
+    ),
+    USUBJID == "01-701-1033" & VSTESTCD == "WEIGHT" ~ case_when(
+      VISIT == "SCREENING 1" ~ 5.62,
+      VISIT == "BASELINE" ~ 5.89,
+      VISIT == "WEEK 2" ~ 6.17,
+      VISIT == "WEEK 4" ~ 6.45
+    ),
+    USUBJID == "01-701-1033" & VSTESTCD == "HDCIRC" ~ case_when(
+      VISIT == "SCREENING 1" ~ 40.34,
+      VISIT == "BASELINE" ~ 40.71,
+      VISIT == "WEEK 2" ~ 41.12,
+      VISIT == "WEEK 4" ~ 41.56
+    ),
+    USUBJID == "01-701-1034" & VSTESTCD == "HEIGHT" ~ case_when(
+      VISIT == "SCREENING 1" ~ 53.65,
+      VISIT == "BASELINE" ~ 54.78,
+      VISIT == "WEEK 2" ~ 55.95,
+      VISIT == "WEEK 4" ~ 57.14,
+      VISIT == "WEEK 6" ~ 58.23,
+      VISIT == "WEEK 8" ~ 59.45,
+      VISIT == "WEEK 12" ~ 60.67,
+      VISIT == "WEEK 16" ~ 61.79,
+      VISIT == "WEEK 20" ~ 62.91,
+      VISIT == "WEEK 24" ~ 64.03,
+      VISIT == "WEEK 26" ~ 65.15
+    ),
+    USUBJID == "01-701-1034" & VSTESTCD == "WEIGHT" ~ case_when(
+      VISIT == "SCREENING 1" ~ 4.25,
+      VISIT == "BASELINE" ~ 4.56,
+      VISIT == "WEEK 2" ~ 4.84,
+      VISIT == "WEEK 4" ~ 5.15,
+      VISIT == "WEEK 6" ~ 5.47,
+      VISIT == "WEEK 8" ~ 5.79,
+      VISIT == "WEEK 12" ~ 6.11,
+      VISIT == "WEEK 16" ~ 6.43,
+      VISIT == "WEEK 20" ~ 6.75,
+      VISIT == "WEEK 24" ~ 7.07,
+      VISIT == "WEEK 26" ~ 7.39
+    ),
+    USUBJID == "01-701-1034" & VSTESTCD == "HDCIRC" ~ case_when(
+      VISIT == "SCREENING 1" ~ 38.45,
+      VISIT == "BASELINE" ~ 38.92,
+      VISIT == "WEEK 2" ~ 39.34,
+      VISIT == "WEEK 4" ~ 39.78,
+      VISIT == "WEEK 6" ~ 40.23,
+      VISIT == "WEEK 8" ~ 40.65,
+      VISIT == "WEEK 12" ~ 41.07,
+      VISIT == "WEEK 16" ~ 41.48,
+      VISIT == "WEEK 20" ~ 41.89,
+      VISIT == "WEEK 24" ~ 42.30,
+      VISIT == "WEEK 26" ~ 42.72
+    ),
+    TRUE ~ VSSTRESN
+  )) %>%
+  arrange(USUBJID, VISITNUM, VSTESTCD, VSDY) %>%
+  # vs_subset_calc <- vs_subset_full
+
+  group_by(USUBJID, VISITNUM) %>%
```

Comment:
i would make this a separate step with its own dataframe and comment associated to explain that BMI vaues are being derived from weight/height, as breaking up the code into clear blocks makes it more modular and easier to maintain in future if ever needing updates. also you could try using https://pharmaverse.github.io/admiral/reference/derive_param_bmi.html here if you like from admiral to derive the BMI.

Reply:
Thank you for the comment. Fully agree. The code has been broken down into functional sections with appropriate comments for BMI calculation. I have, however, stayed with the previously coded derivation. Unless there was a simpler way to do that, derive_param_bmi required me to rename existing VSTESTCD and VSSTRESN, or create a set of working variables like PARAMCD and AVAL, and drop prepared placeholder records for the BMI parameter. 
Should you find it more appropriate to use the function rather than keeping existing code I will do that. Please advise. Thank you.

Reply:
thats fine Remi!